lib/fileutils.rb

```diff
@@ -501,25 +501,81 @@ def cp_lr(src, dest, noop: nil, verbose: nil,
+  #     File.symlink?('destdir1/src1.txt') # => true
+  #
+  # - When +dest+ is the path to an existing file,
+  #   creates a symbolic link at +dest+ pointing to +src+
+  #   if and only if keyword argument <tt>force: true</tt> is given
+  #   (raises an exception otherwise):
+  #
+  #     FileUtils.touch('src2.txt')
+  #     FileUtils.touch('dest2.txt')
+  #     FileUtils.ln_s('src2.txt', 'dest2.txt', force: true)
+  #     FileTest.symlink?('dest2.txt') # => true
+  #
+  #     FileUtils.ln_s('src2.txt', 'dest2.txt') # Raises Errno::EEXIST.
+  #
+  # When +src+ and +dest+ are both paths to directories,
+  # creates a symbolic link at <tt>dest/src</tt> pointing to +src+:
+  #
+  #   FileUtils.mkdir('srcdir3')
+  #   FileUtils.touch('srcdir3/src1.txt')
+  #   FileUtils.touch('srcdir3/src2.txt')
+  #   FileUtils.mkdir('destdir3')
+  #   FileUtils.ln_s('srcdir3', 'destdir3')
+  #   File.symlink?('destdir3/srcdir3') # => true
   #
-  # In the first form, creates a symbolic link +link+ which points to +target+.
-  # If +link+ already exists, raises Errno::EEXIST.
-  # But if the <tt>force</tt> option is set, overwrites +link+.
+  # When +src+ is an array of paths to existing files and +dest+ is a directory:
+  #
+  #   FileUtils.mkdir('srcdir4')
+  #   FileUtils.touch('srcdir4/src0.txt')
+  #   FileUtils.touch('srcdir4/src1.txt')
+  #   FileUtils.mkdir('destdir4')
+  #   FileUtils.ln_s(['srcdir4/src0.txt', 'srcdir4/src1.txt'], 'destdir4')
+  #   File.symlink?('destdir4/src0.txt') # => true
+  #   File.symlink?('destdir4/src1.txt') # => true
+  #
+  # Keyword arguments:
+  #
+  # - <tt>force: true</tt> - overwrites +dest+ if it exists.
+  # - <tt>noop: true</tt> - does not create links.
+  # - <tt>verbose: true</tt> - prints an equivalent command:
+  #
+  #     FileUtils.ln_s('src0.txt', 'dest0.txt', noop: true, verbose: true)
+  #     FileUtils.ln_s('src1.txt', 'destdir1', noop: true, verbose: true)
+  #     FileUtils.ln_s('src2.txt', 'dest2.txt', force: true, noop: true, verbose: true)
+  #     FileUtils.ln_s('srcdir3', 'destdir3', noop: true, verbose: true)
+  #     FileUtils.ln_s(['srcdir4/src0.txt', 'srcdir4/src1.txt'], 'destdir4', noop: true, verbose: true)
+  #
+  #   Output:
   #
-  #   FileUtils.ln_s '/usr/bin/ruby', '/usr/local/bin/ruby'
-  #   FileUtils.ln_s 'verylongsourcefilename.c', 'c', force: true
+  #     ln -s src0.txt dest0.txt
+  #     ln -s src1.txt destdir1
+  #     ln -sf src2.txt dest2.txt
+  #     ln -s srcdir3 destdir3
+  #     ln -s srcdir4/src0.txt srcdir4/src1.txt destdir4
   #
-  # In the second form, creates a link +dir/target+ pointing to +target+.
-  # In the third form, creates several symbolic links in the directory +dir+,
-  # pointing to each item in +targets+.
-  # If +dir+ is not a directory, raises Errno::ENOTDIR.
+  # FileUtils.symlink is an alias for FileUtils.ln_s.
   #
-  #   FileUtils.ln_s Dir.glob('/bin/*.rb'), '/home/foo/bin'
+  # Raises an exception if +src+ is not the path to a file or directory.
```

Comment:
There are probably cases where it raises an exception, but you can use this to symlink to a pipe, another symlink, etc..  I would delete this line unless the cases where it actually fails can be enumerated and it makes sense to do so.

Reply:
Deleted.